plugins/inputs/linux_mem/linux_mem.go

```diff
@@ -0,0 +1,20 @@
+// +build linux
+
+package linux_mem
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+func init() {
+	inputs.Add("linux_meminfo", func() telegraf.Input {
+		return Meminfo{}
+	})
+	inputs.Add("linux_buddyinfo", func() telegraf.Input {
+		return Buddyinfo{}
+	})
+	inputs.Add("linux_slabinfo", func() telegraf.Input {
+		return Slabinfo{}
```

Comment:
When I tried to run these plugins I got the error `E! Error parsing telegraf.linux_mem.conf, v must be a pointer or map`, it looks like we need to return the three inputs as pointers.

Reply:
@danielnelson @phemmer Could this be merged if these two issues are resolved or are there more changes to be done?

Reply:
Some of the meminfo fields have been added to the mem input for 1.8, although now I notice that I forgot to document the new fields properly.  What data are you most interested in?

Reply:
I really like the dynamic approach of this input plugin and the idea to expose all available fields in `/proc/meminfo`. If one only wants a subset of the available fields, measurement filtering should be used as discussed in https://github.com/influxdata/telegraf/issues/1990

The `/proc/slabinfo` and `/proc/buddyinfo` inputs added in this PR are a very nice to have too!

Users that are only interested in what gopsutil exposes could choose to not enable this input and stick with `mem`